tests/test_fio_dump.py

```diff
@@ -28,3 +31,13 @@ def test_dump_layer():
         result = runner.invoke(dump.dump, [TESTGPX, '--layer', layer])
         assert result.exit_code == 0
         assert '"FeatureCollection"' in result.output
+
+
+def test_dump_layer_vfs():
+    path = 'zip://tests/data/coutwildrnp.zip'
+    result = CliRunner().invoke(dump.dump, [path])
```

Comment:
Can you make this `CliRunner().invoke(main_group, 'dump', ...)`? That will help with moving the Fiona env to the main_group ctx someday like we did in Rasterio.